dspace-api/src/main/java/org/dspace/identifier/IdentifierServiceImpl.java

```diff
@@ -112,11 +114,99 @@ public void register(Context context, DSpaceObject dso)
         contentServiceFactory.getDSpaceObjectService(dso).update(context, dso);
     }
 
+    @Override
+    public void register(Context context, DSpaceObject dso, Class<? extends Identifier> type, Filter filter)
+            throws AuthorizeException, SQLException, IdentifierException {
+        boolean registered = false;
+        // Iterate all services and register identifiers as appropriate
+        for (IdentifierProvider service : providers) {
+            if (service.supports(type)) {
+                try {
+                    if (service instanceof FilteredIdentifierProvider) {
+                        FilteredIdentifierProvider filteredService = (FilteredIdentifierProvider)service;
+                        filteredService.register(context, dso, filter);
+                    } else {
+                        service.register(context, dso);
+                    }
+                    registered = true;
+                } catch (IdentifierNotApplicableException e) {
+                    log.warn("Identifier not registered (inapplicable): " + e.getMessage());
+                }
+            }
+        }
+        if (!registered) {
+            throw new IdentifierException("Cannot register identifier: Didn't "
+                    + "find a provider that supports this identifier.");
+        }
+        // Update our item / collection / community
+        contentServiceFactory.getDSpaceObjectService(dso).update(context, dso);
+    }
+
+    @Override
+    public void register(Context context, DSpaceObject dso, Class<? extends Identifier> type)
```

Comment:
Any chance to call previous method with null or TrueFilter as "filter" parameter? 
register(Context context, DSpaceObject dso, Class<? extends Identifier> type, Filter filter)

Reply:
No, I don't think so - I understand the idea here, but note that the `service.register(...)` calls here are actually to a provider (handle provider or DOI provider), not this service.
When the identifier service here doesn't receieve any filter parameter, it should not pass anything onto providers. This allows the providers to use their own defaults without being clobbered (eg. there is a DOI filter specifically set in identifier spring config)

Reply:
Ok, got it. Thank you.